simd/src/Kokkos_Simd_Common.hpp

```diff
@@ -0,0 +1,262 @@
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a * b;
+  return a;
+}
+
+template <class T, class Abi>
+KOKKOS_FORCEINLINE_FUNCTION simd<T, Abi>& operator/=(
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a / b;
+  return a;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator+(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) + b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator+(simd<T, Abi> const& a, U const& b) {
+  return a + simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator-(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) - b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator-(simd<T, Abi> const& a, U const& b) {
+  return a - simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator*(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) * b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator*(simd<T, Abi> const& a, U const& b) {
+  return a * simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator/(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) / b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator/(simd<T, Abi> const& a, U const& b) {
+  return a / simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>&>
+    operator/=(simd<T, Abi>& a, U const& b) {
+  a = a / b;
+  return a;
+}
+
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool all_of(bool a) {
+  return a;
+}
+
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool any_of(bool a) {
+  return a;
+}
+
+// fallback implementations of transcendental functions.
+// individual Abi types may provide overloads with more efficient
+// implementations.
+
+template <class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION simd<T, Abi> exp(simd<T, Abi> a) {
+  T a_array[simd<T, Abi>::size()];
+  a.copy_to(a_array, element_aligned_tag());
+  for (int i = 0; i < simd<T, Abi>::size(); ++i) {
```

Comment:
Is `size()` an `int`? Is that what the standard will do?

Reply:
ooops, no. The standard says `size()` returns `std::size_t`. I can change this loop.